show/main.py

```diff
@@ -960,8 +960,16 @@ def version(verbose):
     asic_type = version_info['asic_type']
     asic_count = multi_asic.get_num_asics()
 
-    serial_number_cmd = "sudo decode-syseeprom -s"
-    serial_number = subprocess.Popen(serial_number_cmd, shell=True, text=True, stdout=subprocess.PIPE)
+    serial_number=''
+    db = SonicV2Connector()
+    db.connect(db.STATE_DB)
+    eeprom_table = db.get_all(db.STATE_DB, 'EEPROM_INFO|0x23')
```

Comment:
One concern is whether this will hang if the DB is unconfigured or dead. `show version` should always work regardless of the state of the DB. Can you please confirm that this works and does not hang on an unconfigured switch?

Reply:
> One concern is whether this will hang if the DB is unconfigured or dead. `show version` should always work regardless of the state of the DB. Can you please confirm that this works and does not hang on an unconfigured switch?

The show version command hangs even without my change when "CONFIG_DB_INITIALIZED" is removed from CONFIG_DB. With no docker database the show version command returns a trace. My change doesn't modify any of the existing behavior. Without state-db eeprom table populated, the actual eeprom is accessed and serial number is fetched. This can be verified by stopping pmon container

Reply:
Thanks for testing. As long as the behavior doesn't change with this patch, I'm OK. If we have to revisit the behavior later we'll cross that bridge then.